modules/gateway/peers.go

```diff
@@ -162,6 +163,40 @@ func (g *Gateway) acceptConn(conn net.Conn) {
 		return
 	}
 
+	// Read the peer's port that we can dial them back on. Peers older than v0.6.1
+	// will only be discovered by newer peers via the ShareNodes RPC.
+	var remoteAddr modules.NetAddress
+	if build.VersionCmp(remoteVersion, "0.6.1") >= 0 {
+		var dialbackPort string
+		err := encoding.ReadObject(conn, &dialbackPort, 13) // Max port # is 65535 (5 digits long) + 8 byte string length prefix
+		if err != nil {
+			conn.Close()
+			g.log.Printf("INFO: could not read remote peer's (%v) port: %v", addr, err)
+			return
+		}
+		if _, err := strconv.Atoi(dialbackPort); err != nil {
+			conn.Close()
+			g.log.Printf("INFO: peer (%v) sent an invalid dialback port: %v", addr, err)
```

Comment:
Generally speaking we don't want to `Printf` messages that can be induced by byzantine peers. In this case, the message can be manipulated by the peer being malicious, therefore the logging should be debug only.  Same for some of the other `log.Print` calls in this PR